app/controllers/proposals_controller.rb

```diff
@@ -148,4 +149,28 @@ def require_waitlisted_or_accepted_state
       redirect_to event_url(@event.slug)
     end
   end
+
+  def require_complete_profile
```

Comment:
There are a few things with this approach that's worth exploring.

The method name doesn't feel right.  It's more of a warning vs a requirement.  Usually require_ before_actions redirect if the requirement is not met.